instrumentation/kafka-streams/src/main/java/brave/kafka/streams/KafkaStreamsTracing.java

```diff
@@ -44,25 +49,27 @@
 public final class KafkaStreamsTracing {
 
   final KafkaTracing kafkaTracing;
+  final Tracer tracer;
+  final Set<String> propagationKeys;
   final TraceContext.Extractor<Headers> extractor;
   final TraceContext.Injector<Headers> injector;
 
   KafkaStreamsTracing(Builder builder) { // intentionally hidden constructor
     this.kafkaTracing = builder.kafkaTracing;
-    this.extractor = kafkaTracing.tracing().propagation().extractor(KafkaStreamsPropagation.GETTER);
-    this.injector = kafkaTracing.tracing().propagation().injector(KafkaStreamsPropagation.SETTER);
+    this.tracer = kafkaTracing.messagingTracing().tracing().tracer();
+    Propagation<String> propagation = kafkaTracing.messagingTracing().tracing().propagation();
+    this.propagationKeys = new LinkedHashSet<>(propagation.keys());
+    this.extractor = propagation.extractor(KafkaStreamsPropagation.GETTER);
+    this.injector = propagation.injector(KafkaStreamsPropagation.SETTER);
   }
 
   public static KafkaStreamsTracing create(Tracing tracing) {
-    return new KafkaStreamsTracing.Builder(KafkaTracing.create(tracing)).build();
+    return create(KafkaTracing.create(tracing));
   }
 
+  /** @since 5.9 */
   public static KafkaStreamsTracing create(KafkaTracing kafkaTracing) {
-    return new KafkaStreamsTracing.Builder(kafkaTracing).build();
-  }
-
-  public KafkaTracing kafkaTracing() {
```

Comment:
no use of this so I removed this. In general, we expose things public upon use case, not before.